### PR TITLE
Add MP and gptq support in quick-start

### PR DESCRIPTION
**Pull request by Idan-BenAmi**

### CODE REVIEW

File: tutorials/quick_start/keras_fw/quant.py [hunk 1]
```diff
@@ -66,12 +90,51 @@ def quantize(model: tf.keras.Model,
...
+
+    # Quantize model
+    if args.get('gptq', False):
+
+        technique = 'GPTQ'
+        n_epochs = args.get('gptq_num_calibration_iter') // n_iter
+        logging.info(
+            f"MCT Gradient-based Post Training Quantization is enabled. Number of epochs: {n_epochs}")
+
+        gptq_representative_data_gen = get_representative_dataset(
```
Comment (elad-c) [inline tutorials/quick_start/keras_fw/quant.py:112]: This is irrelevant when _gptq_representative_data_gen_ is identical to _representative_data_gen_. If _gptq_representative_data_gen_ isn't given to the API it defaults to _representative_data_gen_

File: tutorials/quick_start/keras_fw/quant.py [hunk 2]
```diff
@@ -66,12 +90,51 @@ def quantize(model: tf.keras.Model,
...
+                                                                              core_config=core_conf,
+                                                                              gptq_config=gptq_conf,
+                                                                              gptq_representative_data_gen=gptq_representative_data_gen,
+                                                                              target_platform_capabilities=tpc)
+
+
+    else:
+        technique = 'PTQ'
+        quantized_model, quantization_info = \
+            mct.ptq.keras_post_training_quantization_experimental(model,
```
Comment (elad-c) [inline tutorials/quick_start/keras_fw/quant.py:134]: once you remove _gptq_representative_data_gen_ you can call the API outside the if-else

File: tutorials/quick_start/pytorch_fw/quant.py [hunk 3]
```diff
@@ -65,11 +89,51 @@ def quantize(model: nn.Module,
...
-                                                                representative_data_gen=representative_data_gen,
-                                                                target_platform_capabilities=tpc)
+    if args.get('gptq', False):
+
+        technique = 'GPTQ'
+        n_epochs = args.get('gptq_num_calibration_iter') // n_iter
+        logging.info(
+            f"MCT Gradient-based Post Training Quantization is enabled. Number of epochs: {n_epochs}")
+
+        gptq_representative_data_gen = get_representative_dataset(
```
Comment (elad-c) [inline tutorials/quick_start/pytorch_fw/quant.py:111]: same as keras

**Review by elad-c — CHANGES_REQUESTED** [review]
In general add comment with links to the relevant readme files in the github

File: tutorials/quick_start/keras_fw/quant.py [hunk 4]
```diff
@@ -66,12 +90,51 @@ def quantize(model: tf.keras.Model,
...
+            n_iter=n_iter,
+            batch_size=int(args[BATCH_SIZE])
+        )
+
+        gptq_conf = mct.gptq.get_keras_gptq_config(n_epochs=n_epochs)
+
+
+        quantized_model, quantization_info = \
+            mct.gptq.keras_gradient_post_training_quantization_experimental(model,
+                                                                              representative_data_gen=representative_data_gen,
```
Comment (ofirgo) [inline tutorials/quick_start/keras_fw/quant.py:123]: align with the previous line

File: tutorials/quick_start/pytorch_fw/quant.py [hunk 5]
```diff
@@ -39,6 +41,26 @@ def get_tpc(target_platform_name: str, target_platform_version: str) -> TargetPl
     return mct.get_target_platform_capabilities('pytorch', target_platform_name, target_platform_version)
 
 
+def get_target_kpi(model, weights_compression, representative_data_gen, core_config, tpc):
```
Comment (ofirgo) [inline tutorials/quick_start/pytorch_fw/quant.py:44]: All comments regarding text, arguments names and documentation are relevant here as well

**Review by ofirgo — CHANGES_REQUESTED** [review]

File: tutorials/quick_start/common/results.py [hunk 6]
```diff
@@ -45,7 +45,8 @@ class QuantInfo:
     """
     def __init__(self, user_info: UserInformation,
                  tpc_info: dict,
-                 technique: str
+                 technique: str,
```
[the remark this answers is not in the document]
Comment (Idan-BenAmi) [reply tutorials/quick_start/common/results.py:48]: I think quantization method can be confusing, also refer to power of 2 Vs. Uniform,,,

[the remark this answers is not in the document]
Comment (ofirgo) [reply tutorials/quick_start/common/results.py:48]: @Idan-BenAmi 
You see, I thought that was what you meant by "technique"...
So it isn't a good name

File: tutorials/quick_start/common/results.py [hunk 7]
```diff
@@ -109,7 +112,9 @@ def parse_results(params: dict, float_acc: float, quant_acc: float, quant_info:
     """
     a_bits = quant_info.tpc_info['Target Platform Model']['Default quantization config']['activation_n_bits']
     w_bits = quant_info.tpc_info['Target Platform Model']['Default quantization config']['weights_n_bits']
-    bit_config = 'W' + str(w_bits) + 'A' + str(a_bits)
+    bit_config = f'W{str(w_bits)}A{str(a_bits)}'
```
Comment (ofirgo) [inline tutorials/quick_start/common/results.py:115]: I don't think that you need the `str()` here if you used `f' '` form, it does the casting automatically

File: tutorials/quick_start/keras_fw/quant.py [hunk 8]
```diff
@@ -40,15 +43,35 @@ def get_tpc(target_platform_name: str, target_platform_version: str) -> TargetPl
...
+        core_config (CoreConfig): CoreConfig containing parameters for quantization and mixed precision.
+        tpc (TargetPlatformCapabilities): TargetPlatformCapabilities to optimize the TensorFlow model according to.
+
+    Returns:
+        A KPI object computed from MCT and contains info about the target model size.
+
+    """
+    kpi_data = mct.core.keras_kpi_data_experimental(model, representative_data_gen, core_config=core_config, target_platform_capabilities=tpc)
+    weights_kpi = BYTES_TO_FP32 * kpi_data.weights_memory / weights_compression # (4 bytes for fp32) * weights memory(in Bytes) / compression rate
+    return KPI(weights_kpi)
```
Comment (ofirgo) [inline tutorials/quick_start/keras_fw/quant.py:63]: you might want to do:
`KPI(weights_memory=weights_kpi)`
since KPI object gets other arguments as well, and in the future you might want to pass them as well

File: tutorials/quick_start/README.md [hunk 9]
```diff
@@ -37,24 +37,31 @@ In this example, we are running the main.py script with the following parameters
...
-|---------------|---------------|---------------|-------------------------------|-----------------------------------|
-| mobilenet_v2  | torchvision   | ImageNet      | /path/to/validation/dataset   | /path/to/representative/dataset   |
-| regnetx_002   | timm          | ImageNet      | /path/to/validation/dataset   | /path/to/representative/dataset   |
+Please note that the above example assumes the command is run from the command line and that the [main.py](./main.py) script is in the current directory.
 
+Make sure to refer to the script itself to understand all available parameters and their usage.
 
+#### Advanced model quantization example
+##### Mixed-precision 
+In this example, we employ the MCT Mixed-Precision quantization workflow to achieve a more aggressive reduction in the model's size.
```
Comment (ofirgo) [inline tutorials/quick_start/README.md:46]: "employ" == "apply" ?
Or is "employ" correct in this context?

Comment (ofirgo) [inline tutorials/quick_start/README.md:46]: Also, not sure about the word "aggressive".
Maybe say something like "to further reduce the model's size, with minimal reduction in the quantized model accuracy"

File: tutorials/quick_start/README.md [hunk 10]
```diff
@@ -37,24 +37,31 @@ In this example, we are running the main.py script with the following parameters
...
-| regnetx_002   | timm          | ImageNet      | /path/to/validation/dataset   | /path/to/representative/dataset   |
+Please note that the above example assumes the command is run from the command line and that the [main.py](./main.py) script is in the current directory.
 
+Make sure to refer to the script itself to understand all available parameters and their usage.
 
+#### Advanced model quantization example
+##### Mixed-precision 
+In this example, we employ the MCT Mixed-Precision quantization workflow to achieve a more aggressive reduction in the model's size. 
+We use the same pre-trained mobilenet_v2 model as our baseline, with the goal of attaining a model size
+that is 1/5 of the original floating-point model size. This is equivalent to a size reduction factor of 5.
```
Comment (ofirgo) [inline tutorials/quick_start/README.md:48]: refer to it as "total weights size" or "model weights size" instead of "model size"

File: tutorials/quick_start/README.md [hunk 11]
```diff
@@ -37,24 +37,31 @@ In this example, we are running the main.py script with the following parameters
...
-Please note that the above example assumes the command is run from the command line and that the [main.py](./main.py) script is in the current directory.
+You can execute the following Python script to compress the model:
+```python
+python main.py --model_name mobilenet_v2 --model_library torchvision --mp_weights_compression 5 --validation_dataset_folder <my path> --representative_dataset_folder <my path> 
+```
 
-Make sure to refer to the script itself to understand all available parameters and their usage.
+##### Gradient-based post training quantization 
+The following example demontrates the use of MCT's Gradient-based Post-Training Quantization (GPTQ) workflow. 
+This approach is superior to the basic PTQ method as it modifies the quantized model weights for enhanced performance
```
Comment (ofirgo) [inline tutorials/quick_start/README.md:58]: It is a very long sentence that is hard to follow. Consider splitting it

**Review by ofirgo — APPROVED** [review]

**Review by elad-c — APPROVED** [review]
